tls/s2n_cipher_preferences.c

```diff
@@ -90,8 +90,10 @@ struct s2n_cipher_suite *cipher_suites_20140601[] = {
     &s2n_rsa_with_aes_128_cbc_sha256,
     &s2n_rsa_with_aes_128_cbc_sha,
     &s2n_rsa_with_3des_ede_cbc_sha,
+#ifdef LIBCRYPTO_SUPPORTS_EVP_RC4
     &s2n_rsa_with_rc4_128_sha,
     &s2n_rsa_with_rc4_128_md5
+#endif /* LIBCRYPTO_SUPPORTS_EVP_RC4 */
```

Comment:
None of the if/defs should leave [evp_rc4.c](https://github.com/aws/s2n-tls/pull/3301/files#diff-e96fa37bc68acbf0db32c5e7231fb0a3a109b3fb377b448bec123789c717ee22).

When rc4 isn't available, the cipher struct is_available method should return false. All the other methods can be stubs. Basically, create a stub cipher.